### ISSUE TITLE
Windows Workstation DNS Configuration

### ISSUE THREAD

**Issue by mit-jamie-clayton** [hashicorp/consul#3964]
### Description of the Issue (and unexpected/desired result)
As a windows software developer I want to be able to configure consult on my windows workstation to confirm the behaviour of consul independently of any existing DNS infrastructure, so I can confirm the develop windows software services behind consul.

As a windows software developer, I want to read the full set of instructions for using windows and consul without knowing too much about DNS configuration or Linux DNS  command lines need to setup a basic server implementation. 

At the moment the consul documentation is oriented at Linux environments and associated command lines, while the promoting the service as cross platform. In the on premise/cloud/hybrid world I need to understand how consult fits into those models for customer who have chosen Windows infrastructure to host software.

### Reproduction steps
1. On a windows 10 enterprise workstation (or VM) install consult via https://chocolatey.org/packages/consul
2. Create consul.json and save it to C:\ProgramData\consul\config\consul.json
3. Edit the consul.json file and add the following
{
    "server": true,    
    "ui": true, 
    "datacenter": "JamiesLaptopDC",
    "acl_master_token": "thiscanbeanystring",
	"data_dir": "C:\\%ProgramData%\\consul\\config",
    "encrypt": "/hZH8NScRAUq883dImYGrg==",
    "disable_keyring_file": true
}
3. Open powershell as administrator and confirm the configuration
4. PS> consul validate consul.json
5. Confirm the IP address for the service
6. PS>Ipconfig /all
7. Get the appropriate IP address to bind against
8.PS> consul agent -data-dir=C:\ProgramData\consul\config\ -config-file='consul.json' -bind #.#.#.# -bootstrap
9. Confirm the http gui runs via command line or open a web browser and navigate to http://localhost:8500/ui/
10. PS> consul agent -ui 
11. Confirm the dns resolution is working
12. PS> nslookup
13. > consul.service.consul
14. Confirm the DNS resolution on a windows works as expected.

Review the following windows powershell commands
`
# show rules
netsh interface portproxy show all

# add tcp rule for consul - this persists between reboots
netsh interface portproxy add v4tov4 listenport=53 listenaddress=0.0.0.0 connectport=8600 connectaddress=127.0.0.1

# remove rule
netsh interface portproxy delete v4tov4 listenport=53 listenaddress=0.0.0.0

# Review active ports on windows
netstat -abno | Findstr 53

# Try to force DNS lookup to consul
NSlookup
> set port=8600
> server 127.0.0.1
>consul.service.consul

`

### `consul version` for both Client and Server
Client: `1.0.3`
Server: `1.0.3`

### `consul info` for both Client and Server
Server:
```
Consul agent running!
           Version: 'v1.0.3'
           Node ID: '2c5a3f73-1ad5-6607-9dcb-a2232c98fbb9'
         Node name: 'WC060504'
        Datacenter: 'jamieslaptopdc' (Segment: '<all>')
            Server: true (Bootstrap: true)
       Client Addr: [127.0.0.1] (HTTP: 8500, HTTPS: -1, DNS: 8600)
      Cluster Addr: 10.1.64.138 (LAN: 8301, WAN: 8302)
           Encrypt: Gossip: true, TLS-Outgoing: false, TLS-Incoming: false
```

### Operating system and Environment details
Windows 10 Enterprise 64 bit
Version: 1709
OS Build: 16299.248

### Log Fragments or Link to [gist](https://gist.github.com/)
localhost can't find consul.service.consul: No response from server

https://gitter.im/hashicorp-consul/Lobby - Discussion by @mit-jamie-clayton and @brahama around various commands to get DNS resolution working in the windows world.

**Comment by banks** [comment]
Thanks, we have plans to improve windows friendliness in a few of our guides but this is useful detailed feedback on a painful area right now.

@mit-jamie-clayton do you think this deserves to be in the getting started Guides https://www.consul.io/intro/getting-started/install.html or just in the DNS interface docs: https://www.consul.io/docs/agent/dns.html?

Neither of those offer equivalent advice for running DNS resolves and configuring firewalls for Linux or OS X users currently either as far as I can see (not that they shouldn't). Is there something particular to Windows DNS (which I'm unfamiliar with) that makes it more painful than other OSes? Or just the fact that we only show `dig` commands and no windows equivalent?

Thanks.

**Comment by mit-jamie-clayton** [comment]
@banks Yes the documentation for https://www.consul.io/intro/getting-started/install.html should be expanded to include the explicit steps for Windows engineers. I would argue that there are two audiences you should consider. 
1. Domain Infrastructure engineers that want to integrate consult into their existing infrastructure.
2. Software engineers who are authoring micro-services & health checks.  Most likely they will have Docker CE and/or Azure Service Fabric running locally as they design the software and determine/test the implementation options for consul.

With regards to the command line examples in https://www.consul.io/docs/agent/dns.html, I would recommend (love!) to see OS specific examples provided. My world is a combination of Linux containers, Windows Servers and Mac OS/Windows development environments and we try to provide our customer platform agnostic solutions.  In the cloud world we can control this, in the Corporate environment we have to leverage the existing infrastructure to complete the task.

**Comment by mit-jamie-clayton** [comment]
@banks I've watched a lot of the hashicorp videos and I really wanted to watch something for first time users, that walked you through install, confirmation and execution of a "Happy Path" for each of the common scenarios. The existing ones are great, but I struggled to learn and understand before I could "sell" the solution to customers.

1. Software engineer - Linux/Mac OS/Windows
2. Infrastructure engine - On premise Linux/Windows environments
3. Cloud engineer - Aws, Azure, Google
4. Hybrid On Premise and Cloud environments.

**Comment by banks** [comment]
Great feedback thanks.

We have folks who are focussed on easing the learning curve and producing
good materials for our products in the near future so this is timely.

I'll make sure they see the suggestions!

**Comment by ASxa86** [comment]
What's the status on this?

I am a desktop application and simulation developer (C++) attempting to make use of micro-service for the first time. I have no experience with DNS server configuration and there are no examples, that I can find, on the web site for Windows 10 DNS server configuration to point windows to the localhost:8600 dns server.

I am attempting to configure consul with two duplicate micro-services: one on system A and one on System B. I want to be able to access these services through the same dns so when one goes down, the user doesn't notice a thing.

Even the information provided here doesn't appear to be working for me. I ran `netsh interface portproxy add v4tov4 listenport=53 listenaddress=0.0.0.0 connectport=8600 connectaddress=127.0.0.1` but it doesn't appear I can ping my agent or service using a dns name still.

Is there any information that someone can point me to in order to get this up and running?

Thank you.
